src/evo/providertx.h

```diff
@@ -203,45 +263,72 @@ class CProUpRegTx
     static constexpr auto SPECIALTX_TYPE = TRANSACTION_PROVIDER_UPDATE_REGISTRAR;
     static constexpr uint16_t LEGACY_BLS_VERSION = 1;
     static constexpr uint16_t BASIC_BLS_VERSION = 2;
+    static constexpr uint16_t MULTI_PAYOUT_VERSION = 3;
 
-    [[nodiscard]] static constexpr auto GetVersion(const bool is_basic_scheme_active) -> uint16_t
+    [[nodiscard]] static constexpr auto GetVersion(const bool is_basic_scheme_active, const bool is_multi_payout_active) -> uint16_t
     {
-        return is_basic_scheme_active ? BASIC_BLS_VERSION : LEGACY_BLS_VERSION;
+        if (is_multi_payout_active) {
+            // multi payout is activated after basic scheme
+            assert(is_basic_scheme_active);
+            return MULTI_PAYOUT_VERSION;
+        } else {
+            return is_basic_scheme_active ? BASIC_BLS_VERSION : LEGACY_BLS_VERSION;
+        }
     }
 
     uint16_t nVersion{LEGACY_BLS_VERSION}; // message version
     uint256 proTxHash;
     uint16_t nMode{0}; // only 0 supported for now
     CBLSLazyPublicKey pubKeyOperator;
     CKeyID keyIDVoting;
-    CScript scriptPayout;
+    std::vector<PayoutShare> payoutShares;
     uint256 inputsHash; // replay protection
     std::vector<unsigned char> vchSig;
 
-    SERIALIZE_METHODS(CProUpRegTx, obj)
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action)
```

Comment:
I changed to SerializationOp because in this way I could differentiate between reading and writing to have backward compatibility when reading a message with an older version:

if deserializing with a message with an older  version message just read a CScript:
`

    CScript payoutScript;
    READWRITE(payoutScript);
    payoutShares = {PayoutShare(payoutScript)};        
`

if serializing a message with an older version write the first element of the array only:

`
     READWRITE(payoutShares[0].scriptPayout);
`

am I missing something?

Reply:
hm, does [this way](https://github.com/knst/dash/commit/65ab6511515b3cf9ee598e694133a7520cfedfd4) do same thing or I am missing something?

Reply:
We can have a default value for member rewardShare as 10.000 then